persist/fs/seek.go

```diff
@@ -181,6 +181,13 @@ func (s *seeker) Open(namespace ts.ID, shard uint32, blockStart time.Time) error
 		return err
 	}
 
+	if !s.keepUnreadBuf {
```

Comment:
yeah I felt it's nicer to be moved to Open since it's not strongly related to readIndex and this way we don't necessarily have to do readInfo first and then readIndex.